src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -369,9 +370,16 @@ def statfs(self, path):
         device = mountpoint.as_posix().removeprefix('/mnt/')
         device = device.removeprefix('/cluster/')
 
-        # we only look for /mnt/ or /cluster/ paths and, currently,
-        # those 2 paths are limited to zfs and/or fuse.glusterfs
-        fstype = 'zfs' if path.startswith('/mnt/') else 'fuse.glusterfs'
+        # get fstype for given path based on major:minor entry in mountinfo
+        stx = stat_x.statx(path)
+        maj_min = f'{stx.stx_dev_major}:{stx.stx_dev_minor}'
+        fstype = None
+        with open('/proc/self/mountinfo') as f:
+            for line in f:
+                fline = line.split()
+                if len(fline) >= 3 and fline[2] == maj_min:
+                    fstype = line[line.find('-'):].split()[1]
```

Comment:
It would have been nice if you could include a comment containing an example that should qualify to extract fstype from..

Reply:
I'll add a comment on what the line looks like in `/proc/self/mountinfo` 😄